### DISCUSSION TITLE
Error when trying to write into the mysql database

### DISCUSSION THREAD

**Discussion by NourDh** [getredash/redash#6551] [General]
Hello,
I know that Redash was no made and does not really support writing into the database, but, since it is something that was working, some of our team members want that feature, but, we've found out that there is an error of some sort when we try to update any Mysql database (no issue for PostgreSQL), the command executes and we get this error : `"Error running query: No data was returned."` this is probably not an error since Redash is not designed to output anything for these types of queries, but that is what we get whenever we try to update a database, we get that error and nothing changes, we are using the latest version and the Redash user has permissions to write into the database, any idea about this ?

**Comment by justinclift** [comment]
Hi @NourDh, which version of Redash are you trying with?

Aside from that - since I don't use MySQL personally - I'd probably check to see if the query your people are running can be adjusted to return some kind of info.

Along the lines of:

```
UPDATE foo SET bar = 1 RETURNING 0;
```

The `RETURNING` phrase there is a SQL thing that some databases support, and it means the database will return a value to the caller.  In the Redash case, it would return `0` to Redash who would then *probably* continue operating normally without displaying an error.

I haven't tried it personally, thus the *probably* above.  Pretty sure it'd work though. :smile:

**Reply by NourDh** [reply]
Thank you for your reply @justinclift ,
We are currently using the latest version v10.1.0 (redash/redash:10.1.0.b50633) Under kubernetes.
I have tried your idea, but I had to modify it since it didn't work, I replaced `RETURNING` with `SELECT`, so the final query was as follows:
```
select * from table1
SET name = 'test'
WHERE id = 12;

SELECT 0;
```
But, still, even though I get `0` as an output, nothing changes in the database when I do another `SELECT` to check.

**Reply by justinclift** [reply]
Ahhh well.  Sorry I couldn't be of more help. :frowning:
